runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContext.java

```diff
@@ -307,5 +335,34 @@ public Closeable enterState(ExecutionState newState) {
     public String getWorkItemId() {
       return this.workItemId;
     }
+
+    public ActiveMessageMetadata getActiveMessageMetadata() {
+      return activeMessageMetadata;
+    }
+
+    public Map<String, IntSummaryStatistics> getProcessingTimesByStep() {
```

Comment:
this will pass the reference and allow callers to mutate the map.
unless that's intentional, consider wrapping in `Collections.unmodifiableMap(processingTimesByStep)` or returning an `ImmutableMap`go/java-practices/immutability#collections